java/src/main/java/dojo/supermarket/model/Receipt.java

```diff
@@ -0,0 +1,38 @@
+package dojo.supermarket.model;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class Receipt {
+
+    private final List<ReceiptItem> items = new ArrayList<>();
+    private final List<Discount> discounts = new ArrayList<>();
+
+    public double getTotalPrice() {
+        double total = 0.0;
+        for (ReceiptItem item : items) {
+            total += item.getTotalPrice();
+        }
+        for (Discount discount : discounts) {
+            total += discount.getDiscountAmount();
+        }
+        return total;
+    }
```

Comment:
same as in previous comments - BigDecimal. It will allow also elegant stream usage 
```suggestion
    public BigDecimal getTotalPrice() {
        BigDecimal itemTotal = items.stream()
        .map(ReceiptItem::getTotalPrice)
        .reduce(BigDecimal.ZERO, BigDecimal::add);
        BigDecimal discountTotal = discounts.stream()
        .map(Discount::getDiscountAmount)
        .reduce(BigDecimal.ZERO, BigDecimal::add);
        return itemTotal.add(discountTotal);
    }
```